src/Caching/NullCache.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace FastRoute\Caching;
+
+use Psr\SimpleCache\CacheInterface;
+
+final class NullCache implements CacheInterface
+{
+    /** @inheritDoc */
+    public function get($key, $default = null)
+    {
+        return $default;
+    }
+
+    /** @inheritDoc */
+    public function set($key, $value, $ttl = null): bool
+    {
+        return true;
+    }
+
+    /** @inheritDoc */
+    public function delete($key): bool
+    {
+        return true;
+    }
+
+    public function clear(): bool
+    {
+        return true;
+    }
+
+    /** @inheritDoc */
+    public function getMultiple($keys, $default = null): array
+    {
+        return [];
```

Comment:
This implementation does not satisfy the specification. It should be:

```php
return array_fill_keys($keys, $default);
```

Reply:
Actually, you also must throw an exception if keys is not iterable.

```php
if (!is_iterable($keys)) {
    throw new \Psr\SimpleCache\InvalidArgumentException();
}

return array_fill_keys($keys, $default);
```

Reply:
Surely following a Null design pattern it doesn't need to do anything? It's a stand in for empty functionality, so it has to do only the most basic functionality, as a dummy test double would do in unit testing. The function expects a simple return type, so it returns that type to prevent a Type error. Nothing more.

I think you're overthinking things here.

Reply:
I like this pull and want it to be merged asap ;)
Confronting PSR exception throwing (using fastroute paradigm) and php7 return type error :)
As the framework does not prevaults I think we're pointing the same question as the next one, "handle the error handler?" (simplefilecache l.148 : set, include, restore, are you hiding something).
All that for a useless method...